cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1272,6 +1374,16 @@ void writer::impl::write(table_view const& table)
                  "Error in getting compressed size from nvcomp");
   }
 
+  // Find which partition a rg belongs to
+  std::vector<int> part_end_rg;
+  std::inclusive_scan(
+    num_rg_in_part.begin(), num_rg_in_part.end(), std::back_inserter(part_end_rg));
+  std::vector<int> rg_to_part;
+  auto it = thrust::make_counting_iterator(0);
+  std::transform(it, it + num_rowgroups, std::back_inserter(rg_to_part), [&](auto i) {
+    return std::upper_bound(part_end_rg.begin(), part_end_rg.end(), i) - part_end_rg.begin();
+  });
```

Comment:
This is neat but it bugs me that it turns an O(n) problem into O(n logn). I think a for loop over `num_rg_in_part` + `std::fill` would also work, minus the coolness factor. Not a request, but a suggestion.

Reply:
That's how you know I've been parallel programming for too long